### PR TITLE
Expose interfaces in `pyface.api`

### LINKED ISSUE #702: pyface.api should expose interface classes
For example, `pyface.api` does not re-export the `IGUI` interface class but it exports the `GUI` implementation/provider.

### LINKED ISSUE #736: Interfaces aren't part of pyface.api
...but perhaps they shouldn't be, since in many cases when you want the interface you explicitly don't want the implementation (which may end up with side-effects, such as toolkit selection).

However it does mean that there is no one place were interfaces are collected, so classes that want to use them have to import directly from the modules where they are defined.

Perhaps the solution is a standard place for interfaces (eg. `pyface.interfaces`) where public interfaces can be collected for use.

### PR DESCRIPTION
**Pull request by corranwebster**
Fixes #702

I'd sort of like to have a separate "interfaces API" module (`pyface.interfaces` or `pyface.i_api`?) where these can be imported from without side-effects, but this is good enough for now.

### CODE REVIEW

File: pyface/api.py [hunk 1]
```diff
@@ -45,6 +45,36 @@
...
+- :class:`~.IApplicationWindow`
+- :class:`~.IClipboard`
+- :class:`~.IConfirmationDialog`
+- :class:`~.IDialog`
+- :class:`~.IDirectoryDialog`
+- :class:`~.IDropHandler`
+- :class:`~.IFileDialog`
+- :class:`~.IGUI`
+- :class:`~.IHeadingText`
+- :class:`~.IImage`
```
Comment (mdickinson) [inline pyface/api.py:61]: I'm not seeing where `IImage` is imported.

File: pyface/api.py [hunk 2]
```diff
@@ -45,6 +45,36 @@
...
+- :class:`~.IPILImage`
+- :class:`~.IPythonEditor`
+- :class:`~.IPythonShell`
+- :class:`~.IProgressDialog`
+- :class:`~.ISingleChoiceDialog`
+- :class:`~.ISplashScreen`
+- :class:`~.ISplitWidget`
+- :class:`~.ISystemMetrics`
+- :class:`~.IWindow`
+- :class:`~.IWidget`
```
Comment (mdickinson) [inline pyface/api.py:76]: Alphabetization nitpick: `'IWidget' < 'IWindow'`

**Review by mdickinson — APPROVED** [review]
`IImage` appears to be missing from the actual imports. LGTM otherwise

File: pyface/api.py [hunk 3]
```diff
@@ -45,6 +45,36 @@
...
+- :class:`~.IImage`
+- :class:`~.IImageResource`
+- :class:`~.ILayeredPanel`
+- :class:`~.ILayoutItem`
+- :class:`~.ILayoutWidget`
+- :class:`~.IMessageDialog`
+- :class:`~.IPILImage`
+- :class:`~.IPythonEditor`
+- :class:`~.IPythonShell`
+- :class:`~.IProgressDialog`
```
Comment (mdickinson) [inline pyface/api.py:70]: 🔤

**Comment by mdickinson** [comment]
Even with a separate `pyface.interfaces`, I think it would make sense to _also_ expose the interfaces in `pyface.api`.

**Comment by corranwebster** [comment]
> Even with a separate `pyface.interfaces`, I think it would make sense to _also_ expose the interfaces in `pyface.api`.

Agreed, which is why I made this PR and punted on the separate module for now (mainly because naming is hard - I have it implemented).

**Pull request merged**